llvm/lib/Support/Parallel.cpp

```diff
@@ -113,7 +113,7 @@ class ThreadPoolExecutor : public Executor {
 
 private:
   bool hasSequentialTasks() const {
-    return !WorkQueueSequential.empty() && !SequentialQueueIsLocked;
+    return !WorkQueueSequential.empty() && (getThreadIndex() == 0);
```

Comment:
IIUC, this change makes `hasSequentialTasks()` dependent on the thread index and `hasSequentialTasks()` is used as part of the `Cond.wait()` predicate in the worker thread main loop in `work()`. The `add()` method calls `Cond.notify_one()` to "wake" a single waiting worker thread. If the only outstanding task is sequential and the thread that is woken is not thread 0, then couldn't this possibly lead to the task never being executed? I think in practise because of "spurious wakeups", this would not happen but can that be guaranteed?

Reply:
excuse me for the delay. I am working on this issue. 

Yes, we can change the implementation to guarantee that no tasks would be missed. Will submit changes.